### PR TITLE
Add metrics for syslog input

### PR DESCRIPTION
**Pull request by purbon**
Added metrics for:

* # events.
* # messages received.
* # connection attends
* # errors.

related to https://github.com/elastic/logstash/issues/5607 https://github.com/elastic/logstash/issues/5717

### CODE REVIEW

File: lib/logstash/inputs/syslog.rb [hunk 1]
```diff
@@ -136,6 +138,7 @@ def udp_listener(output_queue)
 
     while !stop?
       payload, client = @udp.recvfrom(9000)
+      metric.increment(:messages_received)
```
Comment (jordansissel) [inline lib/logstash/inputs/syslog.rb]: I am feeling like `messages_received` isn't the right word here. Maybe `packets` ?

Let's also include bytes (`payload.bytesize`) as `bytes` metric maybe? Thoughts

File: lib/logstash/inputs/syslog.rb [hunk 2]
```diff
@@ -118,6 +119,7 @@ def server(protocol, output_queue)
   rescue => e
     if !stop?
       @logger.warn("syslog listener died", :protocol => protocol, :address => "#{@host}:#{@port}", :exception => e, :backtrace => e.backtrace)
+      @metric_errors.increment(:listener_exception)
```
Comment (jordansissel) [inline lib/logstash/inputs/syslog.rb]: I think @ph mentioned elsewhere an idea to namespace exceptions, and I agree.

I don't think we should call them exceptions: we should call them errors. Thoughts @ph and @purbon?

`{ "errors" { "listener" => 123 } }`

File: lib/logstash/inputs/syslog.rb (hunk 1 above)
Comment (jordansissel) [inline lib/logstash/inputs/syslog.rb]: Ignore this comment, I am wrong about this.

File: lib/logstash/inputs/syslog.rb [hunk 3]
```diff
@@ -245,6 +254,7 @@ def syslog_relay(event)
       event.set("priority", 13)
       event.set("severity", 5)   # 13 & 7 == 5
       event.set("facility", 1)   # 13 >> 3 == 1
+      metric.increment(:unkown_messages)
```
Comment (jordansissel) [inline lib/logstash/inputs/syslog.rb]: typo `unknown_messages`

Comment (purbon) [inline lib/logstash/inputs/syslog.rb]: good catch, thanks Jordan. will update

File: lib/logstash/inputs/syslog.rb (hunk 2 above)
Comment (purbon) [inline lib/logstash/inputs/syslog.rb]: @ph @jordansissel I think now there are 3 places with similar discussions, well the same. I created https://github.com/elastic/logstash/issues/5716 to centralize, can we agree on pattern there? that would be awesome! then I will make sure to follow it everywhere 😄

Comment (purbon) [inline lib/logstash/inputs/syslog.rb]: @jordansissel as of now the output will be `{ "errors" { "listener_exception" => 123 } }`, but I do like also the idea to remove the `exception` part, lets find consensus with @ph proposals in other PR.

**Comment by purbon** [comment]
@ph @jordansissel PR updated, what do you think now?

**Comment by ph** [comment]
LGTM

**Comment by jordansissel** [comment]
LGTM

**Pull request closed without merging**